client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/survey/survey.tsx

```diff
@@ -0,0 +1,40 @@
+import { useLocale } from '@automattic/i18n-utils';
+import { useState } from 'react';
+import Banner from 'calypso/components/banner';
+import './survey.scss';
+
+const Survey = ( { userHasEngaged }: { userHasEngaged: boolean } ) => {
+	const locale = useLocale();
+	const [ dismissed, setDismissed ] = useState( false );
```

Comment:
We have to move the state to the parent. Otherwise, the user will see the survey again when they go to select a layout or styles and back